lib/chef/resource/windows_task.rb

```diff
@@ -94,22 +97,41 @@ def numeric_value_in_string?(val)
         false
       end
 
+      def validate_frequency_monthly(frequency_modifier, months, day)
+        if frequency_modifier != 1
+          if  frequency_modifier_includes_days_of_weeks?(frequency_modifier)
+            unless day
+              raise ArgumentError, "Please select day on which you want to run the task e.g. 'Mon, Tue'. Multiple values should be commas seprated."
+            end
+          end
+        end
+      end
+
+      def frequency_modifier_includes_days_of_weeks?(frequency_modifier)
+        frequency_modifier.split(',').each do |value|
+          return false unless %w(FIRST SECOND THIRD FOURTH LAST LASTDAY).include?(value.strip.upcase)
+        end
+        true
+      end
+
       def validate_random_delay(random_delay, frequency)
-        if [:once, :on_logon, :onstart, :on_idle, :none].include? frequency
-          raise ArgumentError, "`random_delay` property is supported only for frequency :minute, :hourly, :daily, :weekly and :monthly"
+        if [:on_logon, :onstart, :on_idle, :none].include? frequency
+          raise ArgumentError, "`random_delay` property is supported only for frequency :once, :minute, :hourly, :daily, :weekly and :monthly"
         end
 
         raise ArgumentError, "Invalid value passed for `random_delay`. Please pass seconds as an Integer (e.g. 60) or a String with numeric values only (e.g. '60')." unless numeric_value_in_string?(random_delay)
       end
 
       # @todo when we drop ruby 2.3 support this should be converted to .match?() instead of =~f
       def validate_start_day(start_day, frequency)
-        if [:once, :on_logon, :onstart, :on_idle, :none].include? frequency
+        if start_day && [:on_logon, :onstart, :on_idle, :none].include?(frequency)
           raise ArgumentError, "`start_day` property is not supported with frequency: #{frequency}"
         end
 
         # make sure the start_day is in MM/DD/YYYY format: http://rubular.com/r/cgjHemtWl5
-        raise ArgumentError, "`start_day` property must be in the MM/DD/YYYY format." unless /^(0[1-9]|1[012])[- \/.](0[1-9]|[12][0-9]|3[01])[- \/.](19|20)\d\d$/ =~ start_day
+        if start_day
```

Comment:
I this this should be a method for valid date format
```
require 'date'
def date_valid?(start_day)
  format = '%m-%d-%Y'
  DateTime.strptime(date, format)
  true
rescue ArgumentError
   "`start_day` property must be in the MM/DD/YYYY format."
end
```

Reply:
I don't think this is required here since we are checking the string format since of date coming as string "mm/dd/yyyy" format and it's already working code.